oshmem/mca/scoll/basic/scoll_basic_barrier.c

```diff
@@ -561,7 +561,7 @@ static int _algorithm_adaptive(struct oshmem_group_t *group, long *pSync)
         int i = 0;
 
         for (i = 0; i < group->proc_count; i++) {
-            if (i == group->id)
+            if (i == oshmem_proc_group_find_id(group, group->my_pe))
```

Comment:
oshmem_proc_group_find_id() - can it be called outside loop from line 563?
it seems constant